File: frontend/lib/features/diet/models/meal_type.dart
/// Eating-event categories used to organise meals (backend `MealType`).
enum MealType {
  breakfast('BREAKFAST'),
  lunch('LUNCH'),
  dinner('DINNER'),
  snack('SNACK'),
  other('OTHER');

  const MealType(this.wireName);

  /// The exact value sent to / received from the backend.
  final String wireName;

  String get label {
    return switch (this) {
      MealType.breakfast => 'Breakfast',
      MealType.lunch => 'Lunch',
      MealType.dinner => 'Dinner',
      MealType.snack => 'Snack',
      MealType.other => 'Other',
    };
  }

  static MealType fromWire(String? value) {
    return MealType.values.firstWhere(
      (type) => type.wireName == value,
      orElse: () => MealType.other,
    );
  }
}

File: backend/src/main/java/com/blistra/diet/domain/MealType.java
package com.blistra.diet.domain;

/**
 * Eating-event categories used to organise meals.
 *
 * <p>{@code OTHER} supports free-form meals while {@link Meal#getTitle()} provides
 * the user-chosen name whenever the fixed taxonomy is not a good fit.</p>
 */
public enum MealType {
    BREAKFAST,
    LUNCH,
    DINNER,
    SNACK,
    OTHER
}